sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Helpers/SyntaxHelper.cs

```diff
@@ -163,5 +164,39 @@ public static bool IsAnyKind(this SyntaxToken syntaxToken, ICollection<SyntaxKin
         {
             return collection.Contains((SyntaxKind)syntaxToken.RawKind);
         }
+
+        public static bool ContainsMethodInvocation(BaseMethodDeclarationSyntax methodDeclarationBase, SemanticModel semanticModel,

```

Comment:
I have an idea about this method, but we can implement it in a separate ticket. Basically, we could create a few overloads, depending on whether we need the syntax predicate or the symbol predicate.